### PR TITLE
review tests

### LINKED ISSUE #80: Doublecheck tests
### Bug description

In an earlier PR I came across some strange values in the test data. Therefore I included TODOs in 

- test_zve.py, 
https://github.com/iza-institute-of-labor-economics/gettsim/blob/0323de5170658415d2dba08694d0645c716ba213/gettsim/tests/test_zve.py#L82

- test_arbeitslosengeld_ui.py
https://github.com/iza-institute-of-labor-economics/gettsim/blob/0323de5170658415d2dba08694d0645c716ba213/gettsim/tests/test_arbeitslosengeld_ui.py#L52

- test_tax_sched.py
https://github.com/iza-institute-of-labor-economics/gettsim/blob/0323de5170658415d2dba08694d0645c716ba213/gettsim/tests/test_tax_sched.py#L51

We need to doublecheck the test data if this is just due to rounding errors, due to false formulas calculating the data or errors in the functions.

### PR DESCRIPTION
**Pull request by Eric-Sommer**
### What problem do you want to solve?

closes #80.

**Three tests were reviewed:**
1. `test_ui`: There is one failure due to a deviation of < 0.01€. I think we can accept this.
2. `test_zve`: The reason for deviation is when and how vorsorgeaufwendungen are rounded. I'm reviewing these tests in #43 anyway, therefore I'd rather avoid the double effort.
3. `tax_sched` tests were indeed partly flawed and are now corrected. 
    a. 2009 tax schedule parameters were wrong in the test. Beware! Even though the law was changed in 03/2009, it became effective for the whole tax year. So `2009-01-01` as date is fine. 
    b. `tu` value were calculated wrongly for the first three rows in the test.
    c. There is infact a different `soli`calculation for income tax and capital income tax. Tests and code now account for this.
    d. I had to set `check_dtype=False`because depending on how you convert `ods`to`csv`, integers may become floats or you get an error because `int32 != int64`.

### CODE REVIEW

File: gettsim/taxes/calc_taxes.py [hunk 1]
```diff
@@ -15,32 +15,40 @@ def tax_sched(tax_unit, tb):
     """
 
     adult_married = (~tax_unit["child"]) & (tax_unit["zveranl"])
-    # create ts dataframe and copy three important variables
+
     for inc in tb["zve_list"]:
-        tax_unit["tax_" + inc] = tb["tax_schedule"](tax_unit["zve_" + inc], tb)
+        # apply tax tariff, round to full Euro amounts
+        tax_unit["tax_" + inc] = tb["tax_schedule"](tax_unit["zve_" + inc], tb).astype(
```
Comment (MaxBlesch) [inline gettsim/taxes/calc_taxes.py]: Use f string instead!

File: gettsim/tests/test_arbeitslosengeld_ui.py [hunk 2]
```diff
@@ -49,5 +49,5 @@ def test_ui(input_data, tax_policy_data, year):
         out_cols=[OUT_COL],
         func_kwargs={"tb": tb},
     )
-    # TODO: THis should be reviewed.
+
```
Comment (MaxBlesch) [inline gettsim/tests/test_arbeitslosengeld_ui.py]: Just add a comment, that we verified the correctness and this is due to rounding in ods!

**Comment by hmgaudecker** [comment]
Down the road: let us please not do any rounding, anywhere. I know we deviate slightly from the law there, but the cost for calculating numerical standard errors can be enormous.

**Comment by Eric-Sommer** [comment]
> Down the road: let us please not do any rounding, anywhere. I know we deviate slightly from the law there, but the cost for calculating numerical standard errors can be enormous.

so what now? fix it for now and remove rounding all over the place later on?

**Comment by MaxBlesch** [comment]
ODS is rounding. I think we can't prevent that! We could write the output data from the functions now in the csv, as we now that these values are correct!

File: gettsim/taxes/calc_taxes.py [hunk 3]
```diff
@@ -70,7 +78,7 @@ def abgeltung(tax_unit, tb):
                 tax_unit["gross_e5_tu"] - 2 * (tb["spsparf"] + tb["spwerbz"]), 0
             )
         )
-    return tax_unit_abgelt["abgst"]
+    return tax_unit_abgelt["abgst"].round(2)
```
Comment (MaxBlesch) [inline gettsim/taxes/calc_taxes.py]: As HMG pointed out, we shouldn't round in our code. Sorry didn't saw that in the first round!

File: gettsim/taxes/calc_taxes.py [hunk 4]
```diff
@@ -128,4 +135,4 @@ def soli_formula(solibasis, tb):
         np.maximum(0.2 * (solibasis - tb["solifreigrenze"]), 0),
     )
 
-    return soli
+    return soli.round(2)
```
Comment (MaxBlesch) [inline gettsim/taxes/calc_taxes.py]: Same. No rounding in the code

**Comment by hmgaudecker** [comment]
I would be okay leaving it as is for now. But do not introduce new rounding.

Maybe somebody could open an issue that implements removal of all rounding.

> ODS is rounding. I think we can't prevent that

As per the discussion @mjbloemer, @Eric-Sommer and myself had over Zulip (starting about [here](https://gettsim.zulipchat.com/#narrow/stream/212222-general/topic/test.20data)), this is not a great way to come up with expected results in the first place.

Once #79 is merged and we implement the downstream consequences (not every function takes the whole tb monster but an almost-minimal subset, ...), this should become much easier to do.

**Comment by MaxBlesch** [comment]
Okay. I'll open an issue to remove rounding. I agree that this is not the best way to calculate the test data and in some near future we can also get rid of the ods files!

**Pull request merged**